test/scripts/hexo/hexo.ts

```diff
@@ -218,26 +218,26 @@ describe('Hexo', () => {
     hexo.extend.filter.unregister('before_generate', validateThemeConfig);
   });
 
-  async function testWatch(path) {
-    const target = join(path, 'test.txt');
+  async function testWatch(path, fileName) {
+    const target = join(path, fileName);
     const body = 'test';
     const newBody = body + body;
 
     loadAssetGenerator();
 
     await writeFile(target, body);
     await hexo.watch();
-    await checkStream(route.get('test.txt'), body); // Test for first generation
+    await checkStream(route.get(fileName), body); // Test for first generation
     await writeFile(target, newBody); // Update the file
     await Promise.delay(300);
-    await checkStream(route.get('test.txt'), newBody); // Check the new route
+    await checkStream(route.get(fileName), newBody); // Check the new route
     hexo.unwatch(); // Stop watching
     await unlink(target); // Delete the file
   }
 
-  it('watch() - source', async () => await testWatch(hexo.source_dir));
+  it('watch() - source', async () => await testWatch(hexo.source_dir, 'source.test.watch.txt'));
```

Comment:
I changed it to use unique files for each test for watch. Because using the same files (`test.txt`) as other tests caused failures. Perhaps caused by caching or something similar, but I'm not sure.